frontend/src/components/SignUpPage.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {ChangeEvent, FormEvent, useState} from "react";
+import axios from "axios";
+import {
+    Box,
+    Button,
+    Container,
+    FormControl,
+    InputLabel,
+    Select,
+    SelectChangeEvent,
+    Snackbar,
+    TextField,
+    Typography
+} from "@mui/material";
+import Layout from "./Layout";
+import useAuth from "../hooks/useAuth";
+import MenuItem from "@mui/material/MenuItem";
+
+export default function SignUp() {
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [role, setRole] = useState("");
+    const [confirmPassword, setConfirmPassword] = useState("");
+    const [error, setError] = useState("");
```

Comment:
you could also write them as following: 
    const [username, setUsername] = useState<string>("");
    const [password, setPassword] = useState<string>("");
    const [role, setRole] = useState<string>("");
    const [error, setError] = useState<string>("");